src/v/test_utils/scoped_config.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/base_property.h"
+#include "config/configuration.h"
+
+// Scoped wrapper around config::shard_local_cfg() that tracks get() calls and,
+// upon destructing, resets any properties that were potentially mutated.
+class scoped_config {
+public:
+    ~scoped_config() {
+        for (auto p : _properties_to_reset) {
+            ss::smp::invoke_on_all([p] {
+                config::shard_local_cfg().get(p).reset();
+            }).get();
+        }
+    }
+
+    config::base_property& get(std::string_view name) {
+        _properties_to_reset.emplace_back(name);
+        return config::shard_local_cfg().get(name);
+    }
+
+private:
+    std::list<std::string_view> _properties_to_reset;
```

Comment:
should we save a real string here? as a test author, i think i'm not going to expect to need to manage the lifetime of property strings. for example, wouldn't this be a use-after-free error?

```
scoped.get(std::string("asdf")).something()
```

Reply:
Sure I can do this, thought it's also just not something in our tree atm.

Reply:
ahh i see. yeh, no urgent need then.

Reply:
Oh oops, didn't see the response, but just pushed this change